internet/API_homework.py

```diff
@@ -0,0 +1,75 @@
+import requests
+from bs4 import BeautifulSoup
+import re
+
+
+class GenscanOutput:
+    def __init__(self, status, cds_list, intron_list, exon_list):
+        self.status = status
+        self.cds_list = cds_list
+        self.intron_list = intron_list
+        self.exon_list = exon_list
+
+
+def run_genscan(sequence=None, sequence_file=None, organism="Vertebrate", exon_cutoff=1.00, sequence_name=""):
+    # request
+    url = 'http://hollywood.mit.edu/cgi-bin/genscanw_py.cgi'
+    payload = {
+        "-o": organism,
+        "-e": exon_cutoff,
+        "-n": sequence_name,
+        "-p": "Predicted peptides only"
+    }
+    if sequence is not None:
+        payload['-s'] = sequence
+        resp = requests.post(url, data=payload)
+    if sequence_file is not None:
+        seq_file = open(sequence_file, 'rb')
+        file_list = {'-u': seq_file}
+        resp = requests.post(url, data=payload, files=file_list)
+        seq_file.close()
```

Comment:
На самом деле никто не запрещает использовать `with` и в таких случаях, просто на лекции был не самый удачный пример
```suggestion
        with open(sequence_file, 'rb') as seq_file:
            file_list = {'-u': seq_file}
            resp = requests.post(url, data=payload, files=file_list)
```